src/NuGet.Core/NuGet.Packaging/Signing/Signatures/Signature.cs

```diff
@@ -122,6 +128,37 @@ private static SignatureType GetSignatureType(SignerInfo signer)
             // TODO: Change this to use the new attributes that justin is adding.
             return SignatureType.Author;
         }
+        private static Timestamp GetTimestamp(SignerInfo signer)
+        {
+            var authorUnsignedAttributes = signer.UnsignedAttributes;
+            var timestampCms = new SignedCms();
+
+            foreach (var attribute in authorUnsignedAttributes)
+            {
+                if (string.Equals(attribute.Oid.Value, Oids.SignatureTimeStampTokenAttributeOid))
+                {
+                    timestampCms.Decode(attribute.Values[0].RawData);
+
+                    if (Rfc3161TimestampVerificationUtility.TryReadTSTInfoFromSignedCms(timestampCms, out var tstInfo))
+                    {
+                        const long TicksPerMicrosecond = 10;
+
+                        var accuracy = tstInfo.AccuracyInMicroseconds;
+                        var accuracyTimeSpan = TimeSpan.Zero;
+                        if (accuracy != null)
```

Comment:
The accuracy calculation is missing checking for a certain policy. Check what I was doing here: https://github.com/NuGet/NuGet.Client/blob/dev-patobeltran-sign-verify-messages/src/NuGet.Core/NuGet.Packaging/Signing/Timestamp/Rfc3161TimestampVerificationUtility.cs#L63. Also as a personal preference I don't like to work in ticks, better in milliseconds because is a more standard measurement of time, but if you have a strong preference for ticks feel free to keep it that way. Actually we could merge `Rfc3161TimestampVerificationUtility` with the `Timestamp` class. Most of that code is relevant to the new `Timestamp` object.

Reply:
I didn't care for ticks, I only was trying to get an easy calculation into a new DateTimeOffset. Ticks seemed like the easiest way to not lose precision, but I can do something else if preferred.